### PR TITLE
feat: improve nginx cache conditions

### PR DESCRIPTION
**Pull request by mihai-sysbio**
This PR refactors the use of Docker, according to recent updates. It also revisits the caching rules in Nginx.

### CODE REVIEW

File: nginx/prod.nginx.conf [hunk 1]
```diff
@@ -6,7 +6,7 @@ upstream api {
   server api:8081 weight=10 max_fails=3 fail_timeout=30s;
 }
 
-proxy_cache_path /nginx-cache levels=1:2 keys_zone=my_cache:30m inactive=60m use_temp_path=off max_size=4g;
+proxy_cache_path /nginx-cache levels=2:2 keys_zone=my_cache:300m inactive=1y use_temp_path=off max_size=4g;
```
Comment (e0) [inline nginx/prod.nginx.conf:9]: Why is this changed from `1:2` to `2:2`?

Comment (e0) [reply nginx/prod.nginx.conf:9]: For `16^5` it would need to be `1:2:2` or some permutation of that since only 1 or 2 are allowed. I'm not clear on the optimal number of files per folder so it's difficult for me to say what is enough. Anyways, if there are no performance downsides of having more folders than I don't see why we wouldn't go for the max of `2:2:2`.

Comment (mihai-sysbio) [reply nginx/prod.nginx.conf:9]: Ah right:
> levels of a cache: from 1 to 3, each level accepts values 1 or 2

There performance penalties for traversing more directories, but I'm hoping the machines are now using fast drives. Since `2:2` is not enough, `2:2:2` it is then.

File: nginx/prod.nginx.conf [hunk 2]
```diff
@@ -6,7 +6,7 @@ upstream api {
   server api:8081 weight=10 max_fails=3 fail_timeout=30s;
 }
 
-proxy_cache_path /nginx-cache levels=1:2 keys_zone=my_cache:30m inactive=60m use_temp_path=off max_size=4g;
+proxy_cache_path /nginx-cache levels=2:2:2 keys_zone=my_cache:300m inactive=1y use_temp_path=off max_size=4g;
```
Comment (nanjiangshu) [inline nginx/prod.nginx.conf:9]: If `inactivte=1y` and `max_size=4g`, does it mean when the size of cache exceeds 4 GB, no data will be cached? If so, it would be better to cache recent visits than to keep very old visits in the cache.

Comment (mihai-sysbio) [reply nginx/prod.nginx.conf:9]: How about 10g? It's not enough if all the cache paths are used, but it should still be plenty:
```suggestion
proxy_cache_path /nginx-cache levels=2:2:2 keys_zone=my_cache:300m inactive=1y use_temp_path=off max_size=10g;
```
I'd rather keep the `1y` unless the storage limit is hit.

Comment (mihai-sysbio) [reply nginx/prod.nginx.conf:9]: @nanjiangshu what do you think?

**Review by e0 — APPROVED** [review]
I was waiting for the discussion to resolve but don't think the change would make a breaking difference one way or another so approving this for now.

File: nginx/prod.nginx.conf (hunk 2 above)
Comment (nanjiangshu) [reply nginx/prod.nginx.conf:9]: OK. I think it is fair enough to set max_size to 10g since the root volume of VM is usually not very big. 
I will approve this PR

**Review by nanjiangshu — APPROVED** [review]
